src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -14506,6 +15404,255 @@ public static Vector512<T> Invoke(Vector512<T> x, Vector512<T> y)
             public static Vector512<T> Invoke(Vector512<T> x) => (x * T.CreateChecked(180)) / T.Pi;
         }
 
+        /// <summary>T &lt;&lt; amount</summary>
+        internal readonly struct ShiftLeftOperator<T>(int amount) : IStatefulUnaryOperator<T> where T : IBinaryInteger<T>
+        {
+            private readonly int _amount = amount;
+
+            public static bool Vectorizable => true;
+
+            public T Invoke(T x) => x << _amount;
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public Vector128<T> Invoke(Vector128<T> x)
+            {
+                if (sizeof(T) == 1) return Vector128.ShiftLeft(x.AsByte(), _amount).As<byte, T>();
+                if (sizeof(T) == 2) return Vector128.ShiftLeft(x.AsUInt16(), _amount).As<ushort, T>();
+                if (sizeof(T) == 4) return Vector128.ShiftLeft(x.AsUInt32(), _amount).As<uint, T>();
+                if (sizeof(T) == 8) return Vector128.ShiftLeft(x.AsUInt64(), _amount).As<ulong, T>();
+                throw new NotSupportedException();
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public Vector256<T> Invoke(Vector256<T> x)
+            {
+                if (sizeof(T) == 1) return Vector256.ShiftLeft(x.AsByte(), _amount).As<byte, T>();
+                if (sizeof(T) == 2) return Vector256.ShiftLeft(x.AsUInt16(), _amount).As<ushort, T>();
+                if (sizeof(T) == 4) return Vector256.ShiftLeft(x.AsUInt32(), _amount).As<uint, T>();
+                if (sizeof(T) == 8) return Vector256.ShiftLeft(x.AsUInt64(), _amount).As<ulong, T>();
+                throw new NotSupportedException();
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public Vector512<T> Invoke(Vector512<T> x)
+            {
+                if (sizeof(T) == 1) return Vector512.ShiftLeft(x.AsByte(), _amount).As<byte, T>();
+                if (sizeof(T) == 2) return Vector512.ShiftLeft(x.AsUInt16(), _amount).As<ushort, T>();
+                if (sizeof(T) == 4) return Vector512.ShiftLeft(x.AsUInt32(), _amount).As<uint, T>();
+                if (sizeof(T) == 8) return Vector512.ShiftLeft(x.AsUInt64(), _amount).As<ulong, T>();
+                throw new NotSupportedException();
+            }
+        }
+
+        /// <summary>T &gt;&gt; amount</summary>
+        internal readonly struct ShiftRightArithmeticOperator<T>(int amount) : IStatefulUnaryOperator<T> where T : IBinaryInteger<T>
+        {
+            private readonly int _amount = amount;
+
+            public static bool Vectorizable =>
```

Comment:
You should be able to just use `x >> _amount` and report all types as vectorizable. It treats `unsigned` types the same as `ShiftRightLogical`.

We could also constrain this to `where T : IBinaryInteger<T>, ISignedNumber<T>`, but that isn't strictly necessary IMO and makes writing generic algorithms a bit harder.